templates/vue/src/components/TapestryApp.vue

```diff
@@ -29,16 +30,29 @@ export default {
     return {
       loading: true,
       viewBox: "2200 2700 1600 1100",
+      currentUser: getCurrentUser(),
     }
   },
   computed: {
-    ...mapState(["nodes", "links", "selection", "settings", "rootId"]),
+    ...mapState(["nodes", "links", "selection", "settings"]),
+
     isSidebarOpen() {
       return Boolean(this.$route.query.sidebar)
     },
     analyticsEnabled() {
       return this.settings.analyticsEnabled
     },
+    isTydeView() {
+      const isAdmin = this.currentUser.roles.some(role => role === "administrator")
+      if (!isAdmin && this.settings.tydeModeEnabled) {
+        const rootNode = this.nodes[this.rootId]
+        const hasEditPermissions = this.currentUser.roles.some(role => {
+          return rootNode.permissions[role].some(premission => premission === "edit")
+        })
+        return !hasEditPermissions
+      }
+      return false
```

Comment:
We already have a function to determine if the user can edit the tapestry that should be used (you may need to import that function from the `wp.js` service file).
```suggestion
      return this.settings.tydeModeEnabled && !wp.canEditTapestry()
```